Automattic-Tracks-iOS/Crash Logging/CrashLogging.swift

```diff
@@ -50,19 +50,27 @@ public class CrashLogging {
         return self
     }
 
+    public func shouldSendEvent() -> Bool {
+        #if DEBUG
+        return UserDefaults.standard.bool(forKey: "force-crash-logging")
+        #else
+        return !dataProvider.userHasOptedOut
+        #endif
+    }
```

Comment:
Initially, I thought that `beforeSend` would be called in all cases but looks like that for capturing envelopes this is skipped. For this reason, I moved out this logic from `beforeSend` and expose it, this way Gutenberg can access it and determine if an event should be sent.